src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -418,7 +424,7 @@ protected function createSubQueryBuilder($select = null)
         $queryBuilder = $this->createQueryBuilder($addJoins)->select($select);
 
         if ($this->user && $this->permission && \array_key_exists($this->permission, $this->permissions)) {
-            $this->addAccessControl(
+            $this->accessControlQueryEnhancer->enhance(
```

Comment:
Would keep the trait and fallback to it if the `accessControlQueryEnhancer` is not set.